dev/tests/integration/testsuite/Magento/Catalog/Model/ProductRepositoryTest.php

```diff
@@ -138,4 +141,28 @@ public function testSaveProductWithGalleryImage(): void
         $this->assertStringStartsWith('/m/a/magento_image', $product->getData('image'));
         $this->assertStringStartsWith('/m/a/magento_image', $product->getData('small_image'));
     }
+
+    /**
+     * Test Product Repository can change(update) "sku" for given product.
+     *
+     * @magentoDataFixture Magento/Catalog/_files/product_simple.php
+     * @magentoDbIsolation enabled
+     * @magentoAppArea adminhtml
+     */
+    public function testUpdateProductSku()
+    {
+        $newSku = 'simple-edited';
+        $productId = Bootstrap::getObjectManager()->get(ProductResource::class)->getIdBySku('simple');
+        $initialProduct = Bootstrap::getObjectManager()->create(Product::class)->load($productId);
+
+        $initialProduct->setSku($newSku);
+        $this->productRepository->save($initialProduct);
+
+        $updatedProduct = Bootstrap::getObjectManager()->create(Product::class);
+        $updatedProduct->load($productId);
```

Comment:
Please, avoid using deprecated methods. 
For loading a model, use a resource model and product model factory. So in result we have something like:

```suggestion
    // ProductResource is alias for \Magento\Catalog\Model\ResourceModel\Product
    // ProductFactory is alias for \Magento\Catalog\Model\ProductFactory
    $productFactory =  Bootstrap::getObjectManager()->get(ProductFactory::class)
    $updatedProduct = $productFactory->create()
    $productResource = Bootstrap::getObjectManager()->get(ProductResource::class)
    $productResource->load($updatedProduct, $productId)
```

Reply:
@rogyar updated.